src/txn/txn_rollback_to_stable.c

```diff
@@ -629,16 +774,44 @@ __rollback_abort_row_ondisk_kv(
  */
 static void
 __rollback_abort_newer_col_var(
-  WT_SESSION_IMPL *session, WT_PAGE *page, wt_timestamp_t rollback_timestamp)
+  WT_SESSION_IMPL *session, WT_REF *ref, wt_timestamp_t rollback_timestamp)
 {
+    WT_CELL *kcell;
+    WT_CELL_UNPACK_KV unpack;
     WT_COL *cip;
     WT_INSERT_HEAD *ins;
+    WT_PAGE *page;
+    uint64_t recno, rle;
     uint32_t i;
+    bool stable_update_found;
+
+    page = ref->page;
+    /*
+     * If a disk image exists, start from the provided recno; or else start from 0.
+     */
+    if (page->dsk != NULL)
+        recno = page->dsk->recno;
+    else
+        recno = 0;
+
+    /* Review the changes to the original on-page data items. */
+    WT_COL_FOREACH (page, cip, i) {
+        stable_update_found = false;
 
-    /* Review the changes to the original on-page data items */
-    WT_COL_FOREACH (page, cip, i)
         if ((ins = WT_COL_UPDATE(page, cip)) != NULL)
-            __rollback_abort_newer_insert(session, ins, rollback_timestamp);
+            __rollback_abort_col_newer_insert_update(
```

Comment:
This name is a bit weird. Can it be `__rollback_abort_col_insert_list`?

Reply:
I agree that the update part in the name is unnecessary, how about __rollback_abort_col_newer_insert? I think it would be useful to keep it similar to __rollback_abort_newer_insert() since it serves a similar purpose just for the variable length column store scenario. Otherwise, I could change both functions?

Reply:
I think we should change both.

Reply:
So how about:
__rollback_abort_col_newer_insert_update -> __rollback_abort_col_insert_list
__rollback_abort_newer_insert -> __rollback_abort_insert_list